### ISSUE TITLE
BufferedShuffleDataset: Why does v1.9 removed this class?

### ISSUE THREAD

**Issue by Heepo** [pytorch/pytorch#62310]
Humbly ask, Why does v1.9 removed `BufferedShuffleDataset` ?
I saw that it was introduced in v1.8. This class is helpful when the data is larger than the memory, and we need to do some certain shuffle operations at least inside a buffer. Was it removed or relaced by other API?

cc @VitalyFedyunin @ejguan @SsnL

**Comment by ejguan** [comment]
Hi @Heepo,

I am sorry that the API keeps changing as we are currently rewriting the DataLoader architecture. We plan to use modular `DataPipe` to replace single `Dataset`. The `BufferedShuffle` is moved to https://github.com/pytorch/pytorch/blob/52d1ffb78930c27438b487a9f43e2da47f398401/torch/utils/data/datapipes/iter/combinatorics.py#L48

You can attach it to any existing Dataset to enable the local shuffle.

**Comment by Heepo** [comment]
@ejguan Thanks for your reply and great work! I will try it soon. I hope the new version of pytorch will be consistent with this new DataPipe APi, so that I don't need to change the code of implementing it.

**Comment by ejguan** [comment]
Thanks for your feedback.
We plan to have official release of new DataLoader/DataPipe in the next version of PyTorch. Then, it will become more stable, stay tuned.
